### ISSUE TITLE
Doesn't support TLS

### ISSUE THREAD

**Issue by richardartoul** [tomusdrw/rust-web3#90]
The HTTP transport does not support TLS:

```rust
let (_eloop, transport) = web3::transports::Http::new("https://sokol.poa.network").unwrap();
```

leads to

```bash
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Transport("Io(Error { repr: Custom(Custom { kind: InvalidInput, error: NotHttp }) })")', src/libcore/result.rs:906:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

**Comment by tomusdrw** [comment]
Right, we are using `hyper` directly. We should probably integrate https://docs.rs/hyper-tls/0.1.2/hyper_tls/ under a feature flag.

**Comment by SeunLanLege** [comment]
would like to help out, i'm very familiar with the hyper crate

**Comment by tomusdrw** [comment]
Resolved via #106
